tester/reporter.go

```diff
@@ -199,6 +199,78 @@ func (r JSONCoverageReporter) Report(ch chan *Result) error {
 	return encoder.Encode(report)
 }
 
+// PrettyCoverageReporter reports coverage as text.
+type PrettyCoverageReporter struct {
+	Verbose   bool
+	Cover     *cover.Cover
+	Modules   map[string]*ast.Module
+	Output    io.Writer
+	Threshold float64
+}
+
+// Report prints the test report to the reporter's output.
+func (r PrettyCoverageReporter) Report(ch chan *Result) error {
+	for tr := range ch {
+		if !tr.Pass() {
+			if tr.Error != nil {
+				return tr.Error
+			}
+			return errors.New(tr.String())
+		}
+	}
+	report := r.Cover.Report(r.Modules)
+	if r.Verbose {
+		fmt.Fprintf(r.Output, "Got coverage %v%%\n\n", report.Coverage)
+		for name, coverageReport := range report.Files {
+			fmt.Fprintf(r.Output, "%v %v%%\n", name, coverageReport.Coverage)
+			r.hl()
+			fmt.Fprintln(r.Output, "Covered:")
+			r.printCoverage(name, coverageReport.Covered)
+			if len(coverageReport.NotCovered) > 0 {
+				fmt.Fprintln(r.Output, "Not covered:")
+				r.printCoverage(name, coverageReport.NotCovered)
+			}
+			fmt.Println()
+		}
+	} else {
+		if report.Coverage < r.Threshold {
+			fmt.Fprintln(r.Output, "Missing coverage:")
+			for name, coverageReport := range report.Files {
+				if coverageReport.Coverage != 100 {
+					r.printCoverage(name, coverageReport.NotCovered)
+				}
+			}
+			return &cover.CoverageThresholdError{
+				Coverage:  report.Coverage,
+				Threshold: r.Threshold,
+			}
+		}
+		fmt.Fprintf(r.Output, "Expected coverage %v%%, got %v%%\n", r.Threshold, report.Coverage)
```

Comment:
We shouldn't show "Expected coverage .." when there was no threshold, for example as-is we get:
```
{10:55} ✘ ~/p/g/s/g/o/opa:Syn3rman-coverage-pretty-reporter ✓ ❯ go run . test /tmp/opa-test -c
Expected coverage 0%, got 83.35%
Missing coverage:
/tmp/opa-test/authz.rego: 8
/tmp/opa-test/authz.rego: 10-11
```

We are also showing an empty "Missing Coverage:" section when the coverage is 100%:
```
{11:49} ✘ ~/p/g/s/g/o/opa:Syn3rman-coverage-pretty-reporter ✓ ❯ go run . test /tmp/opa-test -c
Expected coverage 0%, got 100%
Missing coverage:
```

Im also seeing a non-zero exit code when the tests pass and there is no threshold.. which seems odd. I'm assuming thats from the `error("")` being returned?

Most of these are things that should be caught by unit tests, please add some tests to cover this new code.